independent-projects/extension-maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java

```diff
@@ -570,6 +571,16 @@ static String getCodestartArtifact(String originalArtifact, String projectVersio
         return originalArtifact.replace("${project.version}", projectVersion);
     }
 
+    static String getSourceRepo() {
+        // We could try and parse the .git/config file, but that will be fragile
+        // Let's assume we only care about the repo for official-ish builds produced via github actions
+        String repo = System.getenv("GITHUB_REPOSITORY");
```

Comment:
Typically this kind of info can be read directly from POMs. E.g. https://github.com/quarkusio/quarkus/blob/main/pom.xml#L39-L46
Or here is a Micrometer extension parent on Maven Central https://repo1.maven.org/maven2/io/quarkiverse/micrometer/registry/quarkus-micrometer-registry-parent/2.3.0/quarkus-micrometer-registry-parent-2.3.0.pom
Theoretically, you can fetch this info for any Maven artifact. We are doing that in the platform generator. However, it's also not 100% reliable.
What were you going to use this for?

Reply:
It makes sense to add this info to the metadata though. I can help you with that. Ideally, I would like to have a tag recorded as well. Is there a GITHUB_* env var for it?
If we are not building from github actions, we could fallback to Maven POMs for Maven projects. Gradle will be more challenging, I guess.
How about reusing the `scm` term to group this info? E.g.
````
scm
- url
- tag
````

Reply:
I wondered about reading it from the poms, but when I sampled a few, a lot of the quarkiverse extensions were missing the `<scm>` details. For example, https://github.com/quarkiverse/quarkus-jberet/blob/main/pom.xml.) 

As I understand it, the `<scm>` is configured manually, which seems error-prone - unless there's a way we could automate adding it in the CI? Since we always know the scm information when we enrich the yaml, it seemed logical to make a note of what we know, and put in the yaml. 

Once we have the source information we can expose extra information to people to help them choose extensions, such as the level of activity in a project and the number of issues.

Reply:
I think your plan sounds good, @aloubyansky . We could also read the `.git` folder as a fallback, although I would prioritise the pom over that. (And perhaps only implement the folder solution if we're seeing problems, since parsing the `.git` folder may be a lot of work for little reward.)

Looking at the [GitHub environment variables](https://docs.github.com/en/actions/learn-github-actions/environment-variables), I think `GITHUB_REF` will give the tag information. It may need to be interpreted with care since sometimes it's the branch.

Reply:
are we saying we are going to ignore scm info even  if available?

Reply:
I don't think we should ignore it, no. 

In the happy case, the scm information will match what we can tell from `.git` and/or the GitHub actions build environment. 

In other cases, we're not running in GitHub actions/as a git repo, and then we need to rely on the pom for scm information. That's not in the PR yet, but I'll update it to do that. 

The sad case is when the pom says one thing, and the actual run environment says another thing. We'll need to do decide what to do then. Fail the build and make people correct their pom? (Having an accuracy check for the pom would be a nice win.) Decide people know what they're doing and trust the pom, even though it doesn't match reality? Decide reality wins, and report the source that was *actually* built and used for the extension (desirable for security)? Make people set an explicit override flag if the two don't match? 
My preference is for failing the build and not supporting the scenario where reality and the pom don't match, but I don't have strong feelings about it - maybe there there's a legitimate use case for people to put something in the pom that's more aspirational than accurate.

Reply:
The other case to consider is where people manually add the scm information into the yaml, rather than the pom. Clearly, we shouldn't *encourage* this, because it's at best redundant clutter, and at worst inconsistent confusion. But should we support it?

Reply:
If there are multiple sources of SCM found and the info they provide doesn't match, I'd say we should fail the build. We could add an option to log a warning instead but I think it should be an option that would have to be enabled by the user who knows the provided information is misleading.

Reply:
(For a complete answer to your "are we saying we are going to ignore scm info even if available?" question, @maxandersen , I should add that in my first version of the PR I *did* plan to ignore the pom scm info. I have been convinced that this was clearly a bad idea :) )